### PR TITLE
Forge tests for Matmul operator

### PR DESCRIPTION
**Pull request by vobojevicTT**
- Add tests for PyTorch Matmul operator.
- Improve PytestParamsUtils with methods for easy creation of the test data.

### CODE REVIEW

**Comment by vbrkicTT** [comment]
Matmul tests are implemented via lightweight pytest params approach for now. In future we should integrate it via test plan

**Review by vbrkicTT — APPROVED** [review]

File: forge/test/operators/pytorch/matmul/test_matmul.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+# Prepare the tests data which is list of pytest.param() objects.
+# First, create the PytestParamsUtils object that holds the test data.
+utils = PytestParamsUtils()
+
+# fmt: off
+
+# Then use this method to generate the test data.
+# It should be called just once to generate all test cases.
+utils.generate_test_params_list(
```
Comment (dgolubovicTT) [inline forge/test/operators/pytorch/matmul/test_matmul.py:151]: Can we add more detailed comments for this like:

"Generate parameters for testing 3D and 4D input shapes for matmul operations
with different model configurations and data format fidelity levels"

File: forge/test/operators/pytorch/matmul/test_matmul.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    (((1, 3, 3), None, forge.DataFormat.RawUInt8,  forge.MathFidelity.HiFi4),  pytest.mark.xfail(reason=FailingReasons.BUGGY_SHAPE)),   # AssertionError
+    (((1, 3, 3), None, forge.DataFormat.RawUInt32, forge.MathFidelity.HiFi4),  pytest.mark.xfail(reason=FailingReasons.BUGGY_SHAPE)),   # AssertionError
+    (((1, 3, 3), None, forge.DataFormat.RawUInt16, forge.MathFidelity.HiFi4),  pytest.mark.xfail(reason=FailingReasons.BUGGY_SHAPE)),   # AssertionError
+    (((1, 3, 3), None, forge.DataFormat.UInt16,    forge.MathFidelity.HiFi4),  pytest.mark.xfail(reason=FailingReasons.BUGGY_SHAPE)),   # AssertionError
+)
+# fmt: on
+
+
+# Example of how to run just specific case(s) - first skip all, then enable what case to run:
+utils.extend_shape_params_with_marks(
```
Comment (dgolubovicTT) [inline forge/test/operators/pytorch/matmul/test_matmul.py:223]: Do we want to comment this one out? does this call reset marks for all shapes?

**Review by dgolubovicTT — APPROVED** [review]

File: forge/test/operators/pytorch/matmul/test_matmul.py (hunk 1 above)
Comment (vobojevicTT) [reply forge/test/operators/pytorch/matmul/test_matmul.py:154]: I add this comment, tnx!

File: forge/test/operators/pytorch/matmul/test_matmul.py (hunk 2 above)
Comment (vobojevicTT) [reply forge/test/operators/pytorch/matmul/test_matmul.py]: This is an example of how can easily skip all other tests and execute just specified.
I'll comment all, thanks!

File: forge/test/operators/pytorch/matmul/test_matmul.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    def __init__(self):
+        super().__init__()
+        self.testname = "Matmul_operator_test_op_src_from_dram_queue"
+
+
+class ModelConstEvalPass(torch.nn.Module):
+    def __init__(self, shape: TensorShape):
+        super().__init__()
+        self.testname = "Matmul_operator_test_op_src_const_eval_pass"
+        self.c1 = (torch.rand(ShapeUtils.reduce_microbatch_size(shape), requires_grad=False) - 0.5).detach()
```
[the remark this answers is not in the document]
Comment (vbrkicTT) [reply forge/test/operators/pytorch/matmul/test_matmul.py:101]: There is a global set of torch manual seed in https://github.com/tenstorrent/tt-forge-fe/blob/e23bdc1e5a2a67878e3471c4e3a8e8667f5a8fea/forge/test/conftest.py#L86 so tests should be already reproducible.
However while creating torch test input tensors https://github.com/tenstorrent/tt-forge-fe/blob/e23bdc1e5a2a67878e3471c4e3a8e8667f5a8fea/forge/test/operators/utils/compat.py#L251 there is another setting of manual seed via torch generator. Fmpov this looks redundant and should be unified.
